src/spec.c

```diff
@@ -2839,7 +2914,7 @@ void Indexes_UpdateMatchingWithSchemaRules(RedisModuleCtx *ctx, RedisModuleStrin
       continue;
     }
 
-    if (!hashFields || hashFieldChanged(specOp->spec, hashFields)) {
```

Comment:
why the deletion? dead?

Reply:
Not sure why it was added. the `hashFieldChanged` function first checks if `hashFields` is null and returns true if so